src/smt/theory_str_mc.cpp

```diff
@@ -728,7 +728,7 @@ namespace smt {
                 eqc_chars.reset();
                 return true;
             } else {
-                if (termLen.get_unsigned() != iValue.to_string().length()) {
+                if (termLen.get_uint64() != iValue.to_string().length()) {
```

Comment:
I am almost sure, this still will not fix.
The value of "termLen" is decided by the arithmetic solver.
It can be any integer (>= because you add a lower bound, I hope).
So it could get beyond uint64.
The test will for sure be false even then because iValue doesn't fit in reasonable memory. So you want to augment the condition by a check that termLen is an integer (uint64, or even unsigned if you can say more about iValue).

Reply:
I've updated the line you suggested, comparing `rational` to `rational` should work?